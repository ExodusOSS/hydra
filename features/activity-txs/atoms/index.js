export { default as activityTxsAtomDefinition } from './activity-txs'
