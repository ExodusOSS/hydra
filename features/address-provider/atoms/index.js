export { default as addressCacheAtomDefinition } from './address-cache.js'
