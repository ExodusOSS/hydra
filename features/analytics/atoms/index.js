export { default as analyticsUserIdAtomDefinition } from './user-id'
export { default as analyticsExtraSeedsUserIdsAtomDefinition } from './extra-seeds-user-ids'
export { default as analyticsAnonymousIdAtomDefinition } from './anonymous-id'
export { default as shareActivityAtomDefinition } from './share-activity'
export { default as persistedAnalyticsEventsAtomDefinition } from './persistedEvents'
export { default as persistedAnalyticsTraitsAtomDefinition } from './persistedTraits'
