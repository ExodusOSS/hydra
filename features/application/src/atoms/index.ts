import createBackedUpAtom from './backed-up.js'
import { createWalletCreatedAtAtom } from './created-at.js'
import createLockHistoryAtom from './lock-history.js'
import createdLockedAtom from './locked.js'
import createRestoreAtom from './restore.js'
import createAutoLockTimerAtom from './auto-lock-timer.js'
import type { Definition } from '@exodus/dependency-types'

export const lockedAtomDefinition = {
  id: 'lockedAtom',
  type: 'atom',
  factory: createdLockedAtom,
  dependencies: [],
  public: true,
} as const satisfies Definition

export const lockHistoryAtomDefinition = {
  id: 'lockHistoryAtom',
  type: 'atom',
  factory: createLockHistoryAtom,
  dependencies: ['lockedAtom'],
  public: true,
} as const satisfies Definition

export const restoreAtomDefinition = {
  id: 'restoreAtom',
  type: 'atom',
  factory: createRestoreAtom,
  dependencies: [],
  public: true,
} as const satisfies Definition

export const backedUpAtomDefinition = {
  id: 'backedUpAtom',
  type: 'atom',
  factory: createBackedUpAtom,
  dependencies: ['storage'],
  public: true,
} as const satisfies Definition

export const walletCreatedAtAtomDefinition = {
  id: 'walletCreatedAtAtom',
  type: 'atom',
  factory: createWalletCreatedAtAtom,
  dependencies: ['fusion', 'logger'],
  public: true,
} as const satisfies Definition

export const autoLockTimerAtomDefinition = {
  type: 'atom',
  id: 'autoLockTimerAtom',
  factory: createAutoLockTimerAtom,
  dependencies: ['storage'],
  public: true,
} as const satisfies Definition
