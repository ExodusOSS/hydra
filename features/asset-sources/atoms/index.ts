export { availableAssetNamesByWalletAccountAtomDefinition } from './available-asset-names-by-wallet-account.js'
