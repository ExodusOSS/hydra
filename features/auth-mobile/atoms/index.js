import createAuthAtom from './auth'

export const authAtomDefinition = {
  id: 'authAtom',
  type: 'atom',
  factory: createAuthAtom,
  public: true,
}
