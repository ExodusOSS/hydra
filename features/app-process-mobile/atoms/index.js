import createAppProcessAtom from './app-process'
import { createInMemoryAtom } from '@exodus/atoms'

const appProcessAtomDefinition = {
  id: 'appProcessAtom',
  type: 'atom',
  factory: createAppProcessAtom,
  public: true,
}

const appStateHistoryAtomDefinition = {
  id: 'appStateHistoryAtom',
  type: 'atom',
  factory: () => createInMemoryAtom({ defaultValue: [] }),
  public: true,
}

export { appProcessAtomDefinition, appStateHistoryAtomDefinition }
