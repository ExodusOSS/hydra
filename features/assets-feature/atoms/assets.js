import { createInMemoryAtom } from '@exodus/atoms'

const assetsAtomDefinition = {
  id: 'assetsAtom',
  type: 'atom',
  factory: createInMemoryAtom,
  dependencies: [],
  public: true,
}

export default assetsAtomDefinition
