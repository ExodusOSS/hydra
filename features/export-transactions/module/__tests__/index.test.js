import createStorage from '@exodus/storage-memory'
import blockchainMetadataDefinition from '@exodus/blockchain-metadata/module'
import EventEmitter from 'events/'
import { connectAssets } from '@exodus/assets'
import assetsBase from '@exodus/assets-base'
import { OrderSet, PersonalNoteSet, WalletAccount, normalizeTxsJSON } from '@exodus/models'
import { keyBy, mapValues } from '@exodus/basic-utils'
import { createInMemoryAtom } from '@exodus/atoms'
import fixtures from './tx-log-fixtures.json'
import ordersFixtures from './orders-fixtures.json'
import { create as createExportTransactions } from '..'
import {
  accountStatesAtomDefinition,
  txLogsAtomDefinition,
} from '@exodus/blockchain-metadata/atoms'

const assets = connectAssets(assetsBase)

const txFixtures = mapValues(fixtures, (json) => normalizeTxsJSON({ json, assets }))

const { factory: createBlockchainMetadata } = blockchainMetadataDefinition

const createAssetsModule = () =>
  Object.assign(new EventEmitter(), {
    getAssets: () => assets,
    getAsset: (assetName) => assets[assetName],
  })

const personalNoteMessage = 'test'
const personalNotesAtomMock = {
  get: async () =>
    PersonalNoteSet.fromArray([{ txId: txFixtures.solana[0].txId, message: personalNoteMessage }]),
}

describe('export-transactions', () => {
  const walletAccountInstances = [
    WalletAccount.DEFAULT,
    new WalletAccount({ source: 'exodus', index: 1 }),
  ]

  const walletAccountsData = keyBy(walletAccountInstances, (w) => w.toString())

  let enabledWalletAccountsAtom
  let blockchainMetadata
  let assetsModule
  let exportTransactions
  let ordersAtom
  let multipleWalletAccountsEnabledAtom
  let txLogsAtom
  let accountStatesAtom

  beforeEach(async () => {
    const storage = createStorage()
    assetsModule = createAssetsModule()

    enabledWalletAccountsAtom = createInMemoryAtom({
      defaultValue: walletAccountsData,
    })

    ordersAtom = {
      get: async () => OrderSet.fromArray(ordersFixtures),
    }

    multipleWalletAccountsEnabledAtom = createInMemoryAtom({
      defaultValue: true,
    })

    txLogsAtom = txLogsAtomDefinition.factory()
    accountStatesAtom = accountStatesAtomDefinition.factory()

    blockchainMetadata = createBlockchainMetadata({
      assetsModule,
      enabledWalletAccountsAtom,
      storage: storage.namespace('blockchain-metadata'),
      txLogsAtom,
      accountStatesAtom,
    })
    blockchainMetadata.load()

    exportTransactions = createExportTransactions({
      blockchainMetadata,
      assetsModule,
      personalNotesAtom: personalNotesAtomMock,
      enabledWalletAccountsAtom,
      multipleWalletAccountsEnabledAtom,
      ordersAtom,
    })

    await blockchainMetadata
      .batch()
      .updateTxs({
        assetName: 'serum',
        walletAccount: 'exodus_0',
        txs: txFixtures.serum,
      })
      .updateTxs({
        assetName: 'solana',
        walletAccount: 'exodus_0',
        txs: txFixtures.solana,
      })
      .updateTxs({
        assetName: 'raydium',
        walletAccount: 'exodus_0',
        txs: txFixtures.raydium,
      })
      .updateTxs({
        assetName: 'solana',
        walletAccount: 'exodus_1',
        txs: txFixtures.solana2,
      })
      .updateTxs({
        assetName: 'usdcoin_solana',
        walletAccount: 'exodus_0',
        txs: txFixtures.usdcoin_solana,
      })
      .updateTxs({
        assetName: 'bitcoin',
        walletAccount: 'exodus_0',
        txs: txFixtures.bitcoin,
      })
      .commit()
  })

  it('should export and match snapshot', async () => {
    const exportedTxs = await exportTransactions.exportForWalletAccount(
      WalletAccount.DEFAULT.toString()
    )
    expect(exportedTxs).toMatchSnapshot()
  })

  it('should export with personal notes', async () => {
    const exportedTxs = await exportTransactions.exportForWalletAccount(
      WalletAccount.DEFAULT.toString()
    )
    const txWithPersonalNote = exportedTxs.find((tx) => tx.txId === txFixtures.solana[0].txId)

    const personalNotes = await personalNotesAtomMock.get()
    const expectedPersonalNote = personalNotes.get(txFixtures.solana[0].txId).getMessage()
    expect(txWithPersonalNote.personalNote).toMatch(expectedPersonalNote)
  })

  it('should export with order data', async () => {
    const exportedTxs = await exportTransactions.exportForWalletAccount(
      WalletAccount.DEFAULT.toString()
    )

    ordersFixtures.forEach((orderFixture) => {
      const exportedTx = exportedTxs.find((tx) => tx.orderId === orderFixture.orderId)
      expect(exportedTx).toBeDefined()
      expect(exportedTx.type).toEqual('exchange')
    })
  })
})
