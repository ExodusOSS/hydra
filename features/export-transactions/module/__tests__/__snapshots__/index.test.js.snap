// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-transactions should export and match snapshot 1`] = `
[
  {
    "date": 2021-10-27T09:00:43.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": -0.01,
    "outCurrency": "SOL",
    "outTxId": "6SE7MfLC7jWTyP7hDmaCyHH6oxicuhznaPLJyJkvM9pTLAVvQa2cd4QhPXqKARfohp4UosrkzEu3cdjutZP2RBS",
    "outTxUrl": "https://solscan.io/tx/6SE7MfLC7jWTyP7hDmaCyHH6oxicuhznaPLJyJkvM9pTLAVvQa2cd4QhPXqKARfohp4UosrkzEu3cdjutZP2RBS",
    "personalNote": "test",
    "sent": true,
    "toAddress": "CBSn4FsqonSFiXrcDgk5MTMqtHxk9GFWDUPeLyeA2fXd",
    "toPortfolio": "exodus_1",
    "tokens": "",
    "txId": "6SE7MfLC7jWTyP7hDmaCyHH6oxicuhznaPLJyJkvM9pTLAVvQa2cd4QhPXqKARfohp4UosrkzEu3cdjutZP2RBS",
    "type": "withdrawal",
  },
  {
    "date": 2021-10-29T16:10:57.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": -0.01,
    "outCurrency": "SOL",
    "outTxId": "54TNsquA9iS7ioRDeMBwtbzUrp1E9fvNLxkTRVMkxvXjnHNiJ8VRaySQE89WPFJwaD5wfY7CaLvgj6dJGX4mB6Ep",
    "outTxUrl": "https://solscan.io/tx/54TNsquA9iS7ioRDeMBwtbzUrp1E9fvNLxkTRVMkxvXjnHNiJ8VRaySQE89WPFJwaD5wfY7CaLvgj6dJGX4mB6Ep",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
    "toPortfolio": undefined,
    "tokens": "",
    "txId": "54TNsquA9iS7ioRDeMBwtbzUrp1E9fvNLxkTRVMkxvXjnHNiJ8VRaySQE89WPFJwaD5wfY7CaLvgj6dJGX4mB6Ep",
    "type": "withdrawal",
  },
  {
    "date": 2021-11-08T16:01:46.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": -0.005,
    "outCurrency": "SOL",
    "outTxId": "3Kht5MYVdwgG3FsrRDHNneeSyyEeCMt46oGc34Ty3aNLaid5cWxztA3t6BqrxmsnPVHAgjLUW9C9YPi6hYhNtC33",
    "outTxUrl": "https://solscan.io/tx/3Kht5MYVdwgG3FsrRDHNneeSyyEeCMt46oGc34Ty3aNLaid5cWxztA3t6BqrxmsnPVHAgjLUW9C9YPi6hYhNtC33",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "Ua6XTohsCjVdB76tY7QU1fCAirFGJRvDVL82hYRt4bA",
    "toPortfolio": undefined,
    "tokens": "",
    "txId": "3Kht5MYVdwgG3FsrRDHNneeSyyEeCMt46oGc34Ty3aNLaid5cWxztA3t6BqrxmsnPVHAgjLUW9C9YPi6hYhNtC33",
    "type": "withdrawal",
  },
  {
    "date": 2022-04-15T17:08:58.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": 0,
    "outCurrency": "SOL",
    "outTxId": "277NnNEJU6ufZTcNbjQWYYUYDcoXNSG1exBYpL6DJUSxo1BrEykkjYNYPt5sA8vokLFDGaunYnZ8KK5tQRBbMQHr",
    "outTxUrl": "https://solscan.io/tx/277NnNEJU6ufZTcNbjQWYYUYDcoXNSG1exBYpL6DJUSxo1BrEykkjYNYPt5sA8vokLFDGaunYnZ8KK5tQRBbMQHr",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "4BvMcjTP8E6CNnaTp8xHArqYWhdZr2nzTUtFy7naznU5",
    "toPortfolio": undefined,
    "tokens": "",
    "txId": "277NnNEJU6ufZTcNbjQWYYUYDcoXNSG1exBYpL6DJUSxo1BrEykkjYNYPt5sA8vokLFDGaunYnZ8KK5tQRBbMQHr",
    "type": "withdrawal",
  },
  {
    "date": 2022-04-18T16:33:27.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": 0,
    "outCurrency": "SOL",
    "outTxId": "4ebbzyCyx3jpnZ2UmtnhUyEq2JT6UfobvenEC71HtnvoRTbaCLoGwXVRboB1pLe9F9EepRF2AV8xgJ59j24n8Mn3",
    "outTxUrl": "https://solscan.io/tx/4ebbzyCyx3jpnZ2UmtnhUyEq2JT6UfobvenEC71HtnvoRTbaCLoGwXVRboB1pLe9F9EepRF2AV8xgJ59j24n8Mn3",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "4BvMcjTP8E6CNnaTp8xHArqYWhdZr2nzTUtFy7naznU5",
    "toPortfolio": undefined,
    "tokens": "",
    "txId": "4ebbzyCyx3jpnZ2UmtnhUyEq2JT6UfobvenEC71HtnvoRTbaCLoGwXVRboB1pLe9F9EepRF2AV8xgJ59j24n8Mn3",
    "type": "withdrawal",
  },
  {
    "date": 2022-04-21T17:18:48.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": 0,
    "outCurrency": "SOL",
    "outTxId": "v4U1bQ7v6mHczHK4276D6RuVjfN2DaPgc2Gv3tNzYG9c5AbvFGchtHxwcRyFfbNzuY8zdAp2ftQaxPg6CpuuB85",
    "outTxUrl": "https://solscan.io/tx/v4U1bQ7v6mHczHK4276D6RuVjfN2DaPgc2Gv3tNzYG9c5AbvFGchtHxwcRyFfbNzuY8zdAp2ftQaxPg6CpuuB85",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "4BvMcjTP8E6CNnaTp8xHArqYWhdZr2nzTUtFy7naznU5",
    "toPortfolio": undefined,
    "tokens": "",
    "txId": "v4U1bQ7v6mHczHK4276D6RuVjfN2DaPgc2Gv3tNzYG9c5AbvFGchtHxwcRyFfbNzuY8zdAp2ftQaxPg6CpuuB85",
    "type": "withdrawal",
  },
  {
    "date": 2022-05-03T15:49:13.000Z,
    "feeAmount": "",
    "feeCurrency": "",
    "fromPortfolio": undefined,
    "inAmount": 0.396879,
    "inCurrency": "RAY",
    "inTxId": "4wJfB9SJFLxwZcxPdJo8kKwireEJf7LzBztWnHxGAqN5zScSE89fwX5rWxUXe9f2Ub2fsre7bgXHM2EumiaWSECU",
    "inTxUrl": "https://solscan.io/tx/4wJfB9SJFLxwZcxPdJo8kKwireEJf7LzBztWnHxGAqN5zScSE89fwX5rWxUXe9f2Ub2fsre7bgXHM2EumiaWSECU",
    "outAmount": "",
    "outCurrency": "",
    "outTxId": "",
    "outTxUrl": "",
    "personalNote": undefined,
    "sent": false,
    "toAddress": "",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "4wJfB9SJFLxwZcxPdJo8kKwireEJf7LzBztWnHxGAqN5zScSE89fwX5rWxUXe9f2Ub2fsre7bgXHM2EumiaWSECU",
    "type": "deposit",
  },
  {
    "date": 2022-06-13T11:46:49.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": "",
    "inCurrency": "",
    "inTxId": "",
    "inTxUrl": "",
    "outAmount": -0.01,
    "outCurrency": "RAY",
    "outTxId": "4noru63e8XJwEoheyNCnnwmmUMSptsEPTtoJUwSiwP2s5nF9t78QMHtbEXNn4xCN4VMQceMpT2Niv1fPy1ybgrNj",
    "outTxUrl": "https://solscan.io/tx/4noru63e8XJwEoheyNCnnwmmUMSptsEPTtoJUwSiwP2s5nF9t78QMHtbEXNn4xCN4VMQceMpT2Niv1fPy1ybgrNj",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "9xKvRVv7WAfLxcGg9dfcGq6ond8Vkb4VCEkrCCt7BAi6",
    "toPortfolio": undefined,
    "tokens": "",
    "txId": "4noru63e8XJwEoheyNCnnwmmUMSptsEPTtoJUwSiwP2s5nF9t78QMHtbEXNn4xCN4VMQceMpT2Niv1fPy1ybgrNj",
    "type": "withdrawal",
  },
  {
    "date": 2022-09-05T12:48:15.000Z,
    "feeAmount": 0,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": 0.804456,
    "inCurrency": "RAY",
    "inTxId": "2DpWgonfZgQjRz6hfHWwTgV63FBQaF4mUy6RmtUKoLpstUMDMqRMb3pwXRJGJJPcV8cXFBCgbuiYSfsZGJGYvYLD",
    "inTxUrl": "https://solscan.io/tx/2DpWgonfZgQjRz6hfHWwTgV63FBQaF4mUy6RmtUKoLpstUMDMqRMb3pwXRJGJJPcV8cXFBCgbuiYSfsZGJGYvYLD",
    "orderId": "2DpWgonfZgQjRz6hfHWwTgV63FBQaF4mUy6RmtUKoLpstUMDMqRMb3pwXRJGJJPcV8cXFBCgbuiYSfsZGJGYvYLD",
    "outAmount": -0.670595,
    "outCurrency": "SRM",
    "outTxId": "2DpWgonfZgQjRz6hfHWwTgV63FBQaF4mUy6RmtUKoLpstUMDMqRMb3pwXRJGJJPcV8cXFBCgbuiYSfsZGJGYvYLD",
    "outTxUrl": "https://solscan.io/tx/2DpWgonfZgQjRz6hfHWwTgV63FBQaF4mUy6RmtUKoLpstUMDMqRMb3pwXRJGJJPcV8cXFBCgbuiYSfsZGJGYvYLD",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "Go6Qnn5wFKFZoLA9vW29jgGWyNsJ2bJhx2QZyEUGNznx",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "2DpWgonfZgQjRz6hfHWwTgV63FBQaF4mUy6RmtUKoLpstUMDMqRMb3pwXRJGJJPcV8cXFBCgbuiYSfsZGJGYvYLD",
    "type": "exchange",
  },
  {
    "date": 2022-11-01T18:45:08.000Z,
    "feeAmount": "",
    "feeCurrency": "",
    "fromPortfolio": undefined,
    "inAmount": 0.63174,
    "inCurrency": "RAY",
    "inTxId": "2iq3UNWnMQBgSrSaY1xsrxXPxKZf98szWjiK5MsGc2CE7GtKG1qdVNpBSUakwFFbeQ4ZVypxeDhGqN6aoA41VJKK",
    "inTxUrl": "https://solscan.io/tx/2iq3UNWnMQBgSrSaY1xsrxXPxKZf98szWjiK5MsGc2CE7GtKG1qdVNpBSUakwFFbeQ4ZVypxeDhGqN6aoA41VJKK",
    "outAmount": "",
    "outCurrency": "",
    "outTxId": "",
    "outTxUrl": "",
    "personalNote": undefined,
    "sent": false,
    "toAddress": "",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "2iq3UNWnMQBgSrSaY1xsrxXPxKZf98szWjiK5MsGc2CE7GtKG1qdVNpBSUakwFFbeQ4ZVypxeDhGqN6aoA41VJKK",
    "type": "deposit",
  },
  {
    "date": 2022-11-25T18:59:36.000Z,
    "feeAmount": -0.000005,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": 0,
    "inCurrency": "SOL",
    "inTxId": "3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "inTxUrl": "https://solscan.io/tx/3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "orderId": "3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "outAmount": 0,
    "outCurrency": "SOL",
    "outTxId": "3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "outTxUrl": "https://solscan.io/tx/3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "personalNote": undefined,
    "sent": false,
    "toAddress": "",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "type": "exchange",
  },
  {
    "date": 2022-11-25T18:59:36.000Z,
    "feeAmount": "",
    "feeCurrency": "",
    "fromPortfolio": undefined,
    "inAmount": 1.008777,
    "inCurrency": "USDC",
    "inTxId": "3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "inTxUrl": "https://solscan.io/tx/3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "outAmount": "",
    "outCurrency": "",
    "outTxId": "",
    "outTxUrl": "",
    "personalNote": undefined,
    "sent": false,
    "toAddress": "",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "3ZagJbqKzq5YZENUaFn939fz5VE8wBo4sXkCJ99hj9NGxK7aPwRzCpUdEvinbJVH5eyyw6aCRXXG28GJFPUjdLSK",
    "type": "deposit",
  },
  {
    "date": 2022-12-02T19:30:27.000Z,
    "feeAmount": 0,
    "feeCurrency": "SOL",
    "fromPortfolio": "exodus_0",
    "inAmount": 0.331197,
    "inCurrency": "USDC",
    "inTxId": "RUGZKkCCC2WeMoJ2N9uCKCbqKLK2BWp1h3A8YAdYz3E4hGNVMVt7r7NegtPhGS4i3EvhRkNzXvM7UAXn9vdYsFc",
    "inTxUrl": "https://solscan.io/tx/RUGZKkCCC2WeMoJ2N9uCKCbqKLK2BWp1h3A8YAdYz3E4hGNVMVt7r7NegtPhGS4i3EvhRkNzXvM7UAXn9vdYsFc",
    "orderId": "RUGZKkCCC2WeMoJ2N9uCKCbqKLK2BWp1h3A8YAdYz3E4hGNVMVt7r7NegtPhGS4i3EvhRkNzXvM7UAXn9vdYsFc",
    "outAmount": -1.823075,
    "outCurrency": "RAY",
    "outTxId": "RUGZKkCCC2WeMoJ2N9uCKCbqKLK2BWp1h3A8YAdYz3E4hGNVMVt7r7NegtPhGS4i3EvhRkNzXvM7UAXn9vdYsFc",
    "outTxUrl": "https://solscan.io/tx/RUGZKkCCC2WeMoJ2N9uCKCbqKLK2BWp1h3A8YAdYz3E4hGNVMVt7r7NegtPhGS4i3EvhRkNzXvM7UAXn9vdYsFc",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "3h7PhXbCAGvtQHqwTS2V3Mhc3fK8E5Hs8EbgCVHkQFwd",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "RUGZKkCCC2WeMoJ2N9uCKCbqKLK2BWp1h3A8YAdYz3E4hGNVMVt7r7NegtPhGS4i3EvhRkNzXvM7UAXn9vdYsFc",
    "type": "exchange",
  },
  {
    "date": 2023-01-17T17:19:23.102Z,
    "feeAmount": -0.00001193,
    "feeCurrency": "BTC",
    "fromPortfolio": "exodus_0",
    "inAmount": 2.38880897,
    "inCurrency": "SOL",
    "inTxId": "3WDD672f7ZSE1K8ECcs2mWCA85t5N1KenHxJph7k1NG8jbzhzgmQeMoE27Vm8w3yAdCHX8gpD9nRxV5xiZvY3dAF",
    "inTxUrl": "https://solscan.io/tx/3WDD672f7ZSE1K8ECcs2mWCA85t5N1KenHxJph7k1NG8jbzhzgmQeMoE27Vm8w3yAdCHX8gpD9nRxV5xiZvY3dAF",
    "orderId": "pi2xv6mtnnjp5izd",
    "outAmount": -0.0026765,
    "outCurrency": "BTC",
    "outTxId": "c047f99ad171730b94a61c979ffef92b1b93deab5a4aac0cf104e66424d38406",
    "outTxUrl": "https://mempool.space/tx/c047f99ad171730b94a61c979ffef92b1b93deab5a4aac0cf104e66424d38406",
    "personalNote": undefined,
    "sent": true,
    "toAddress": "",
    "toPortfolio": "exodus_0",
    "tokens": "",
    "txId": "c047f99ad171730b94a61c979ffef92b1b93deab5a4aac0cf104e66424d38406",
    "type": "exchange",
  },
]
`;
